src/smolagents/models.py

```diff
@@ -367,6 +375,98 @@ def __call__(
         return message
 
 
+class VLLMModel(Model):
+    """This engine initializes a model and tokenizer from the given `model_id`.
+
+    Parameters:
+        model_id (`str`, *optional*, defaults to `"HuggingFaceTB/SmolLM2-1.7B-Instruct"`):
+            The Hugging Face model ID to be used for inference. This can be a path or model identifier from the Hugging Face model hub.
+    """
+
+    def __init__(self, model_id: Optional[str] = None):
+        super().__init__()
```

Comment:
You likely need to raise an error if VLLM is not here - I would also put the imports in the class to have something more lightweight for non VLLM users